src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptorProvider.cs

```diff
@@ -76,7 +78,9 @@ private void Register(TagHelperDescriptor descriptor)
             if (!_registrations.TryGetValue(descriptor.TagName, out descriptorSet))
             {
                 descriptorSet = new HashSet<TagHelperDescriptor>(TagHelperDescriptorComparer.Default);
-                _registrations[descriptor.TagName] = descriptorSet;
+
+                // We want to register TagHelperDescriptors by their prefix and tag name for easy lookup.
+                _registrations[descriptor.Prefix + descriptor.TagName] = descriptorSet;
```

Comment:
This is the answer to this that I agreed on: https://github.com/aspnet/Razor/pull/313#discussion_r25491125